### DISCUSSION TITLE
Physical total thrust limits of the F450 drone: simulation vs reality

### DISCUSSION THREAD

**Discussion by bconvens** [ctu-mrs/mrs_uav_system#91] [Q&A]
Hi,

My goal is to find the correct physical thrust limit to validate my control law.
From the simulations with the F450 drone, I observe the total thrust limit where the controller saturates to be around 39N. 
This seems to be perfectly in accordance with the [motor bench tests ](https://ctu-mrs.github.io/docs/hardware/motor_tests.html) ctu has performed for the Readytosky 2312 920KV with Aliexpress plastic 9450 self-tightening propeller. According to the table, each motor could approximately take 1kg.

Now I did some manual flight tests where I tried to hover my F450 drone (with px4 firmware of ctu) with fully charged (16.8V) LiPo's and masses rigidly attached to the drone.
It seems that my F450 cannot hover with 3.4kg (total mass) anymore. It takes off, but also immediately goes down again and touches the ground so I have to trigger the emergency switch. 
This thrust is significantly lower as the value used in simulations. In simulation I can increase the mass up to 3.9kg (I changed it in the mass.yaml and xacro) though for whatever motor parameters I put. At 4.0 kg the drone cannot hover anymore. So I am confused. 

Is this 3.4kg limit I observed the real physical limit where all the motors saturate or should it normally still fly with fully charged batteries at 3.4kg? 
If not, then why is there such a huge difference with the motor tests and the simulations? The only physical interpretation I can  see  is the airflow on the F450 arms, but does this mean a loss of 500grams?
Or are there other hardware / software settings I need to check to understand why the difference between simulation and reality is so big. 
Also I wonder, is there any part in the firmware that would limit the steady state thrust (e.g. around 3.4kg) on top of a higher value for the peak thrust (around 4kg)? So I mean some code that checks and chooses a limit based on how long you saturate above the first limit (e.g. at most 5s) or above the second highest limit (e.g. at most 0.5s). 

Thanks for your help! ;-)

Best,

**Comment by DanHert** [comment]
Hello, as we found out, the thrust tests are not completely representative, the performance of the motor/propeller combo is worse when mounted on a real drone, than when it is on the idealized test stand. This is probably due to the interference with the other propellers mounted nearby, and interference with the drone arms/body. The simulation is not perfect, we did not fine tune the thrust parameters to completely correspond to the real world.

Also, our system will emergency-land the drone, if the total required thrust is over 80% for some time. The drone needs headroom for maneuvering, so you cannot fly at 100% thrust and still retain control. 

And I would highly discourage flying at these loads approaching the 80% thrust, it is much better to fly near 50%. If you need more payload, consider getting a bigger drone.

Dan H.

**Reply by bconvens** [reply]
Hello,

Thanks for the answer, that clarifies my questions. 
So for the real drones, there can be an error of about 20% compared to the results reported in [motor bench tests](https://ctu-mrs.github.io/docs/hardware/motor_tests.html).
My idea was not to fly near such high payloads, but I added these payloads only to determine the max physical thrust the drone can produce.

Bryan
